code/modules/research/xenoarchaeology/artifact/artifact_duplicator.dm

```diff
@@ -0,0 +1,188 @@
+/obj/machinery/duplicator
+	name = "alien machine"
+	desc = "It's some kind of large computer. There appear to be disks of some kind anchored to the sides."
+	icon = 'icons/obj/xenoarchaeology.dmi'
+	icon_state = "duplicator_idle"
+	density = 1
+	var/obj/structure/duplicator_pod/leftpod
+	var/obj/structure/duplicator_pod/rightpod
+	var/deployed = FALSE
+	var/duplicating = FALSE
+	var/dupe_time = 10 SECONDS
+
+	idle_power_usage = 100
+	active_power_usage = 1000
+	use_power = 1
+
+	machine_flags = WRENCHMOVE | FIXED2WORK
+
+/obj/machinery/duplicator/New()
+	..()
+	leftpod = new(src)
+	rightpod = new(src)
+	update_icon()
+
+/obj/machinery/duplicator/update_icon()
+	..()
+	icon_state = initial(icon_state)
+	if(stat & NOPOWER || !anchored)
+		icon_state = "duplicator_off"
+	else if(duplicating)
+		icon_state = "duplicator_duplicating"
+
+/obj/machinery/duplicator/power_change()
+	. = ..()
+	update_icon()
+	if(anchored && !deployed)
+		deploy_pods()
+	else if(!anchored && deployed)
+		retract_pods()
+
+/obj/machinery/duplicator/proc/deploy_pods()
+	desc = "It's some kind of large computer."
+	visible_message("The disks detach from the sides of \the [src] and unfold into large pods.")
+	leftpod.forceMove(locate(x-1,y,z))
+	rightpod.forceMove(locate(x+1,y,z))
+	flick("duplicator_pad_deploying", leftpod)
+	flick("duplicator_pad_deploying", rightpod)
+	deployed = TRUE
+
+/obj/machinery/duplicator/proc/retract_pods()
+	desc = initial(desc)
+	visible_message("The pods retract into their bases and reattach to the sides of \the [src].")
+	leftpod.forceMove(src)
+	rightpod.forceMove(src)
+	deployed = FALSE
+
+/obj/machinery/duplicator/attack_hand(mob/user as mob)
+	if(..() || !anchored)
+		return 1
+	interact(user)
+
+/obj/machinery/duplicator/interact(mob/user)
+	if(!leftpod || !rightpod)
+		visible_message("\The [src] buzzes \"<span class='warning'>ERROR: Both pods not present.</span>\"")
+		flick("duplicator_denied", src)
+		return
+	if(!leftpod.get_object() || !rightpod.get_object())
```

Comment:
You call `leftpod.get_object()` about 4 times in this code block, and similarly for the right pod - just store the result of the first call in a var and check that.